### PR TITLE
Use canonical way to check if a map contains a key

### PR DESCRIPTION
**Pull request by HuKeping**
As the language spec:
https://golang.org/ref/spec#Index_expressions

And also fix a typo.

Signed-off-by: Hu Keping <hukeping@huawei.com>

### CODE REVIEW

**Comment by mtrmac** [comment]
:+1:

**Comment by endophage** [comment]
LGTM.

While you're in here it would be great if you add a short test to trigger the `http.Error`. Will let this sit for 24 hours and merge on Friday anyway if you don't have time.

**Comment by diogomonica** [comment]
Agreed on the test, but otherwise LGTM

**Comment by HuKeping** [comment]
Test ready :)

**Comment by HuKeping** [comment]
BTW, I found that we have injected a specific `http.ResponseWriter` object for them to write back the http response. 

Wonder if we could do that (I mean, writing back the http response) outside the `getCryptoService`. Anyway it is not what this PR would like to deal with. If it worth to fix, I'll make another PR some days later.

File: signer/api/api_test.go [hunk 1]
```diff
@@ -225,3 +225,26 @@ func TestSignHandlerReturns404WithNonexistentKey(t *testing.T) {
 
 	assert.Equal(t, 404, res.StatusCode)
 }
+
+func TestCreateKeyHandlerWithInvalidAlgorithm(t *testing.T) {
+	keyStore := trustmanager.NewKeyMemoryStore(passphraseRetriever)
+	cryptoService := cryptoservice.NewCryptoService("", keyStore)
+	setup(signer.CryptoServiceIndex{data.ED25519Key: cryptoService, data.RSAKey: cryptoService, data.ECDSAKey: cryptoService})
+
+	// The `rbtree-algorithm` is expected as not suppoted
```
Comment (cyli) [inline signer/api/api_test.go]: Non-blocking nitpick - "supported" is mispelled.  :)  Minor though, I will merge in morning whether or not it's fixed.

**Comment by cyli** [comment]
LGTM!  Thank you for your contribution!

**Comment by HuKeping** [comment]
updated @cyli  :)

**Comment by cyli** [comment]
@HuKeping thanks for fixing so quickly!!

**Pull request merged**